### PR TITLE
fix: sync cluster level resource status

### LINKED ISSUE #2394: Cluster Resource Binding controller should sync status for resources
**background**
In current code logic, crb controller didn't sync resource status to crd in kramada cluster:
https://github.com/karmada-io/karmada/blob/822abc540b622bad61f1319cc9bbe7d1363fe102/pkg/controllers/binding/cluster_resource_binding_controller.go#L121-L135

If I want to check Resource status like PV, I only go to the member cluster to check.

**What would you like to be added**:
I have added sync status logic in my branch and it's quite same as resource binding controller sync logic:
https://github.com/karmada-io/karmada/blob/303d4652efe70777c1de6b33360f24e7348208e5/pkg/controllers/binding/binding_controller.go#L146-L151

And also need logic to `aggregateStatusInterpreter`. If this fix is acceptable for this repo  I can submit a PR, please let me know if this bugfix/feature is acceptable.

**Why is this needed**:
User need check some cluster level crd status from karmada cluster.

### PR DESCRIPTION
**Pull request by yusank**
Signed-off-by: yusank <yusankurban@gmail.com>

**What type of PR is this?**

kind bug

**What this PR does / why we need it**:

sync cluster scoped resources(like PV) status from member cluster to karmada cluster.

**Which issue(s) this PR fixes**:
Fixes #2394 

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
@yusank Thanks!
/kind feature
Have you tested with it? Do we need to consider also adding the related E2E test?

**Comment by yusank** [comment]
> @yusank Thanks! /kind feature Have you tested with it? Do we need to consider also adding the related E2E test?

I've tested in my local environment and it works . Adding E2E test sounds great idea but I'm not sure how to add it, can give me some example or instruction to add a E2E test ?

**Comment by XiShanYongYe-Chang** [comment]
If add e2e, we can wirte like this: https://github.com/karmada-io/karmada/blob/e237614954a64b2595b9ec18128834a538975518/test/e2e/resourceinterpreter_test.go#L160

We can create a resource just like [Workload](https://github.com/karmada-io/karmada/blob/e237614954a64b2595b9ec18128834a538975518/examples/customresourceinterpreter/apis/workload/v1alpha1/workload_types.go#L13), but a cluster scope resource.

**Comment by RainbowMango** [comment]
Given it's the first contribution from @yusank , E2E would be too complicated.
Let's focus on the code change for this PR.

**Comment by yusank** [comment]
> Given it's the first contribution from @yusank , E2E would be too complicated. Let's focus on the code change for this PR.

Sure @RainbowMango , waiting for code review.

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go [hunk 1]
```diff
@@ -364,6 +365,35 @@ func aggregatePodStatus(object *unstructured.Unstructured, aggregatedStatusItems
 	return helper.ToUnstructured(pod)
 }
 
+func aggregatePersistentVolumeStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (*unstructured.Unstructured, error) {
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go:368]: Can we add a unit test?

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go [hunk 2]
```diff
@@ -364,6 +365,35 @@ func aggregatePodStatus(object *unstructured.Unstructured, aggregatedStatusItems
...
+		if item.Status == nil {
+			continue
+		}
+		temp := &corev1.PersistentVolumeStatus{}
+		if err = json.Unmarshal(item.Status.Raw, temp); err != nil {
+			return nil, err
+		}
+		klog.V(3).Infof("Grab persistentVolume(%s/%s) status from cluster(%s), phase: %s",
+			pv.Namespace, pv.Name, item.ClusterName, temp.Phase)
+		newStatus.Phase = temp.Phase
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go:385]: The `newStatus.Phase` will be overwritten over loops, and only keeps the last `temp.phase`. I don't think this is we expected behavior. 

Maybe you can check how we deal with this similar thing at `aggregatePodStatus()`.

Comment (yusank) [reply pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go:385]: Sure, let me push another commit

Comment (yusank) [reply pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go]: test added

**Comment by yusank** [comment]
sorry for the last two commits, formatted by my IDEA 😞

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go [hunk 3]
```diff
@@ -364,6 +365,59 @@ func aggregatePodStatus(object *unstructured.Unstructured, aggregatedStatusItems
...
 
+func aggregatePersistentVolumeStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (*unstructured.Unstructured, error) {
+	pv := &corev1.PersistentVolume{}
+	err := helper.ConvertToTypedObject(object, pv)
+	if err != nil {
+		return nil, err
+	}
+
+	newStatus := &corev1.PersistentVolumeStatus{}
+	podPhases := sets.NewString()
```
Comment (RainbowMango) [inline pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go:376]:
```suggestion
	volumePhases := sets.NewString()
```
Not `Pod`.

**Review by RainbowMango — COMMENTED** [review]
Just a nit, otherwise LGTM.
Please help squash your commits in next push, thanks.

File: pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go (hunk 3 above)
Comment (yusank) [reply pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go:376]: OK， I'll rebase to one commit

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

Thanks and welcome to Karmada!

**Pull request merged**